repo-agent/issue-sandbox/main.go

```diff
@@ -0,0 +1,185 @@
+		}
+	}()
+
+	err = runIssueSolver()
+	if err != nil {
+		log.Fatalf("failed solving issue: %v", err)
+	}
+
+	err = cmdCodeSrv.Wait()
+	if err != nil {
+		log.Printf("Code Server exited with error: %v", err)
+	} else {
+		log.Println("Code Server exited with no error")
+	}
+}
+
+func runIssueSolver() error {
+	log.Println("Starting issue solver")
+
+	// Environment variables
+	gitPushEnabled := os.Getenv("GIT_PUSH_ENABLED") == "true"
+	githubUserOrigin := os.Getenv("GITHUB_USER_ORIGIN")
+	githubUserLogin := os.Getenv("GITHUB_USER_LOGIN")
+	githubToken := os.Getenv("GITHUB_TOKEN")
+	githubUserEmail := os.Getenv("GITHUB_USER_EMAIL")
+	githubUserName := os.Getenv("GITHUB_USER_NAME")
+	issueBranch := os.Getenv("ISSUE_BRANCH")
+	agentPrompt := os.Getenv("AGENT_PROMPT")
+	issueID := os.Getenv("ISSUEID")
+
+	if _, err := runCommand("git", "remote", "remove", "origin"); err != nil {
+		log.Printf("could not remove origin, probably because it does not exist: %v", err)
+	}
+
+	if gitPushEnabled && githubUserOrigin != "" {
+		originURL := fmt.Sprintf("https://%s:%s@%s", githubUserLogin, githubToken, githubUserOrigin)
+		if _, err := runCommand("git", "remote", "add", "origin", originURL); err != nil {
+			return fmt.Errorf("failed to add origin: %w", err)
+		}
+	}
+
+	if githubUserEmail != "" {
+		if _, err := runCommand("git", "config", "--global", "user.email", githubUserEmail); err != nil {
+			return fmt.Errorf("failed to set git user email: %w", err)
+		}
+	}
+
+	if githubUserName != "" {
+		if _, err := runCommand("git", "config", "--global", "user.name", githubUserName); err != nil {
+			return fmt.Errorf("failed to set git user name: %w", err)
+		}
+	}
+
+	oldCommitID, err := runCommand("git", "rev-parse", "HEAD")
+	if err != nil {
+		return fmt.Errorf("failed to get old commit id: %w", err)
+	}
+
+	if _, err := runCommand("git", "checkout", "-b", issueBranch); err != nil {
+		return fmt.Errorf("failed to create issue branch: %w", err)
+	}
+
+	// Handle .gemini directory
+	if _, err := os.Stat("/workspaces/.gemini"); err == nil {
+		log.Println(".gemini directory exists in /workspaces, copying to repo directory")
+		if _, err := os.Stat(".gemini"); err == nil {
+			log.Println(".gemini directory exists in repo directory, moving to .gemini.bak")
+			if err := os.Rename(".gemini", ".gemini.bak"); err != nil {
+				return fmt.Errorf("failed to move .gemini to .gemini.bak: %w", err)
+			}
+		}
+		if _, err := runCommand("cp", "-R", "/workspaces/.gemini", ".gemini"); err != nil {
+			return fmt.Errorf("failed to copy .gemini directory: %w", err)
+		}
+	} else {
+		log.Println(".gemini directory does not exist in /workspaces")
+	}
+
+	// Run gemini
+	if _, err := os.Stat("../agent-prompt.txt"); os.IsNotExist(err) {
+		log.Println("agent-prompt.txt does not exist, running gemini")
+		if err := os.WriteFile("../agent-prompt.txt", []byte(agentPrompt), 0644); err != nil {
+			return fmt.Errorf("failed to write agent-prompt.txt: %w", err)
+		}
```

Comment:
Splitting the `GEMINI_COMMAND` by spaces is not robust and will fail if arguments contain spaces. It's safer to pass the command and its arguments separately. For example, you could have another environment variable for the arguments or expect the command to be a single executable.